core/src/main/java/org/hibernate/ogm/hibernatecore/impl/BackendCustomLoader.java

```diff
@@ -97,6 +101,14 @@ private boolean isEntityQuery() {
 		}
 	}
 
+	private TransactionContext transactionContext(SessionImplementor session) {
```

Comment:
This seems to exist in several places. How about making it a static method on `TransactionContext` itself.